torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -148,6 +152,11 @@ static Value* tryMatchArgument(
   // also allow single ints/floats to be passed in their place. The single
   // int/float is then repeated to the length of the list
   if (isIntOrFloatUsedAsList(value, arg)) {
+    if (value->type()->kind() == OptionalType::Kind) {
```

Comment:
This feels slightly out of place. Basically the implementation of `isIntOrFloatUsedAsList` "leaked" outside. Maybe we should change behavior of `isIntOrFloatUsedAsList` to `maybeUseIntOrFloatAsList`, which returns an updated value that includes logic that are currently in line 155 to line 162 (even the list creation part).

Reply:
Now changed to `maybeUseIntOrFloatAsList`